itests/fe/permission_requests_test.py

```diff
@@ -1,187 +1,178 @@
+        setup.add_user_to_group("cbguder@a.co", "group-admins")
+        setup.grant_permission_to_group(PERMISSION_ADMIN, "", "group-admins")
+
+    # The old API requires SQLAlchemy objects.
+    granting_user = User.get(setup.session, name="zorkian@a.co")
+    assert granting_user
+    granting_group = Group.get(setup.session, name="auditors")
+    assert granting_group
+    requesting_user = User.get(setup.session, name="cbguder@a.co")
+    assert requesting_user
+    requesting_group = Group.get(setup.session, name="group-admins")
+    assert requesting_group
+    perm_granter = Permission.get(setup.session, "perm.hasgranter")
+    assert perm_granter
+    perm_nogranter = Permission.get(setup.session, "perm.nogranter")
+    assert perm_nogranter
+    perm_admin = Permission.get(setup.session, PERMISSION_ADMIN)
+    assert perm_admin
+
+    # The old APIs require a global settings object.
+    set_global_settings(setup.settings)
+
+    # Request the two test perms from group-admins.
+    with setup.transaction():
+        create_request(
+            setup.session, requesting_user, requesting_group, perm_granter, "a", "reasons"
+        )
+        create_request(
+            setup.session, requesting_user, requesting_group, perm_nogranter, "a", "reasons"
+        )
+
+    # Finally make one more request from a user other than cbguder@a.co.
+    with setup.transaction():
+        create_request(setup.session, granting_user, granting_group, perm_admin, "a", "reasons")
+
+
+def action_permission_requests(setup: SetupTest) -> None:
+    """Action (approve) the perm.hasgranter request, cancel (deny) the perm.nogranter request."""
+    granting_user = User.get(setup.session, name="zorkian@a.co")
+    assert granting_user
+    admin_user = User.get(setup.session, name="cbguder@a.co")
+    assert admin_user
+
+    with setup.transaction():
+        all_requests = setup.session.query(PermissionRequest)
+        for request in all_requests:
+            if request.status == "pending" and request.permission.name == "perm.hasgranter":
+                update_request(setup.session, request, granting_user, "actioned", "reasons")
+            if request.status == "pending" and request.permission.name == "perm.nogranter":
+                update_request(setup.session, request, admin_user, "cancelled", "reasons")
+
+
+def test_pending_inbound_requests(tmpdir: LocalPath, setup: SetupTest, browser: Chrome) -> None:
+    create_permission_requests(setup)
+
+    with frontend_server(tmpdir, "cbguder@a.co") as frontend_url:
+        browser.get(url(frontend_url, "/permissions/requests?status=pending"))
+        page = PermissionRequestsPage(browser)
+
+        # Check the request rows.
+        requested = [row.requested for row in page.request_rows]
+        expected = ["perm.hasgranter, a", "perm.nogranter, a", f"{PERMISSION_ADMIN}, a"]
+        assert sorted(requested) == sorted(expected)
+
+        # Check the status change rows.
```

Comment:
Likely not worth the effort, but it should be feasible to have `PermissionRequestsPage` be smarter about status change rows to avoid this messiness, right? Specifically having `StatusChangeRow`s included in each appropriate `RequestViewRow`

Reply:
Yeah, it's a larger project than I want to tackle right now, but I made a note for myself for later.